### ISSUE TITLE
Bug: [explicit-function-return-type] Allow method implementation

### ISSUE THREAD

**Issue by alecmev** [typescript-eslint/typescript-eslint#5515]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=JYOwLgpgTgZghgYwgAgILIN4ChnIBYQA2hA9gBQCUAXMgM5hSgDmWAvllhAB4AOJUYZAkJxatZACFkwALY9CEGRHDj02XAWLkKmZFAhgArlBDIA5AHd+hACZnk7dkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9IgB6H0nk0BmsDErnIB7BtWiJcsaOILF0kRNGgjokMAF8QmoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
interface A {
  hello(): string
}

export class B implements A {
  hello() { return 'world' }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/explicit-function-return-type": "error",
  },
};
```

### Expected Result

I expected no error.

### Actual Result

I'm asked to put a redundant explicit return type. It's already enforced, try changing `'world'` to `123`.

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.33.1` |
| `@typescript-eslint/parser` | `5.33.1` |
| `TypeScript` | `4.7.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by bradzacher** [comment]
The rule is not (and will never be) type-aware.
Which means we cannot reliably detect which methods are implemented in an interface and which aren't.
